router_test.go

```diff
@@ -59,7 +77,7 @@ func TestRouter(t *testing.T) {
 
 	w := new(mockResponseWriter)
 
-	req, _ := http.NewRequest("GET", "/user/gopher", nil)
+	req := newRequest(t, "GET", "/user/gopher")
```

Comment:
Why did you change everything to `newRequest` instead of `http.NewRequest`, even for unaffected tests?